src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -123,16 +124,18 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 		assertNonNullish(btcAddress, 'No BTC address provided to get BTC transactions.');
 
 		const balance = await this.loadBtcBalance({ identity, bitcoinNetwork });
-		const newTransactions = data?.shouldFetchTransactions
-			? await this.loadBtcTransactions({ btcAddress })
+		const { newTransactions, latestBitcoinBlockHeight } = data?.shouldFetchTransactions
```

Comment:
small thing: would you mind using `nonNullish(data) && data.shouldFetchTransactions` here please? Since we use it like that

Reply:
Of course!